packages/oft-sand/contracts/sand/ERC20BasicApproveExtension.sol

```diff
@@ -17,13 +18,17 @@ abstract contract ERC20BasicApproveExtension is ERC20Internal, Context {
         uint256 amount,
         bytes calldata data
     ) external payable returns (bytes memory) {
-        require(BytesUtil.doFirstParamEqualsAddress(data, _msgSender()), "FIRST_PARAM_NOT_SENDER");
+        if (!BytesUtil.doFirstParamEqualsAddress(data, _msgSender())) {
+            revert FirstParamNotSender();
+        }
 
         _approveFor(_msgSender(), target, amount);
 
         // solhint-disable-next-line avoid-low-level-calls
         (bool success, bytes memory returnData) = target.call{value: msg.value}(data);
```

Comment:
msg.value - still correct?

Reply:
Had this discussion with Maxime in the past. We tried upgrading it but couldn't because of some issue during upgrade, plus these two additional points
1. Sandbox is one with admin role and admin wouldn't be doing gasless transactions.
2. To use _msgSender() here, we will have to upgrade all the other contract inheriting it ( Sand, PolygonSand)